demo/src/main/java/com/novoda/movies/rateable/RateableMovieViewHolder.java

```diff
@@ -0,0 +1,120 @@
+import com.novoda.accessibility.Action;
+import com.novoda.accessibility.Actions;
+import com.novoda.accessibility.ActionsAccessibilityDelegate;
+import com.novoda.accessibility.ActionsAlertDialogCreator;
+import com.novoda.movies.R;
+
+import java.util.Arrays;
+
+class RateableMovieViewHolder extends RecyclerView.ViewHolder {
+
+    static RateableMovieViewHolder inflate(ViewGroup parent) {
+        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
+        View view = layoutInflater.inflate(R.layout.item_view_rateable_movie, parent, false);
+        return new RateableMovieViewHolder(view);
+    }
+
+    private final ImageView posterImageView;
+    private final TextView titleTextView;
+    private final ImageView likeImageView;
+    private final RatingBar ratingBar;
+    private final AccessibilityServices a11yServices;
+
+    RateableMovieViewHolder(View itemView) {
+        super(itemView);
+        posterImageView = itemView.findViewById(R.id.item_rateable_image_poster);
+        titleTextView = itemView.findViewById(R.id.item_rateable_text_title);
+        likeImageView = itemView.findViewById(R.id.item_rateable_image_like);
+        ratingBar = itemView.findViewById(R.id.item_rateable_rating);
+        a11yServices = AccessibilityServices.newInstance(itemView.getContext());
+    }
+
+    void bind(RateableMovieViewModel viewModel) {
+        posterImageView.setImageResource(viewModel.poster);
+        titleTextView.setText(viewModel.title);
+        likeImageView.setImageResource(viewModel.liked ? R.drawable.ic_favorite_24dp : R.drawable.ic_favorite_border_24dp);
+        ratingBar.setRating(viewModel.rating);
+        Actions actions = collateActionsFor(viewModel);
+        ActionsAccessibilityDelegate a11yDelegate = new ActionsAccessibilityDelegate(itemView.getResources(), actions);
+        ViewCompat.setAccessibilityDelegate(itemView, a11yDelegate);
+        itemView.setContentDescription(viewModel.title + ", rating " + viewModel.rating + ", liked: " + viewModel.liked);
+
+        if (a11yServices.isSpokenFeedbackEnabled() || a11yServices.isSwitchAccessEnabled() || !itemView.isInTouchMode()) {
+            bindForIndirectAccess(actions);
+            a11yDelegate.setClickLabel(R.string.action_rateable_movie_usage_hint_click);
+        } else {
+            bindForTouchAccess(viewModel);
+        }
+    }
+
+    private Actions collateActionsFor(RateableMovieViewModel viewModel) {
+        return new Actions(
+                Arrays.asList(
+                        selectActionFor(viewModel),
+                        toggleLikeActionFor(viewModel),
+                        rateActionFor(viewModel)
+                )
+        );
+    }
+
+    private void bindForIndirectAccess(Actions actions) {
+        itemView.setOnClickListener(v -> new ActionsAlertDialogCreator(itemView.getContext()).create(actions).show());
+        likeImageView.setClickable(false);
+        ratingBar.setOnRatingBarChangeListener(null);
```

Comment:
the first line is our "see more options" thing. when the user clicks on the item view, it'll open an alert dialog containing the list of actions that we collated earlier.

one thing I noticed when testing with the switch is that it's annoying to cancel the alert dialog because you have to iterate through all the options, get to the "menu" and then press "back". I wonder if having a "cancel" at the top of the dialog (automatically in the dialog creator, perhaps optional) would be better. Then again, since neither switch access nor talkback remembers the last focused item when a window is closed, the user would have to navigate to that item again from the start of the view hierarchy if they accidentally cancelled, and moreover, having the cancel at the top would be weird as the first item the user gets after explicitly clicking "more options". maybe some user testing around this would be beneficial 💸 

```java
        likeImageView.setClickable(false);
        ratingBar.setOnRatingBarChangeListener(null);
```

these two lines were intended to stop the accessibility services from focusing on them. setting `importantForAccessibility="no"` in the xml achieves the same but I don't want to push and nuke all these comments 😭 also I'm not sure if that attribute trumps a view's clickability or vice versa in _all_ versions of android so I'll leave it for now.

the reason we want the services to ignore these views is because we have our list dialog of actions which takes its place. this means navigating through the list is simpler/quicker - because services will only focus on the itemview as a whole, and not any of the children.